core/trino-main/src/main/java/io/trino/metadata/AbstractPropertyManager.java

```diff
@@ -69,28 +75,191 @@ protected final void doRemoveProperties(K propertiesKey)
+            Iterable<Property> properties,
             Session session,
             Metadata metadata,
             AccessControl accessControl,
-            Map<NodeRef<Parameter>, Expression> parameters,
-            boolean setDefaultProperties)
+            Map<NodeRef<Parameter>, Expression> parameters)
     {
-        Map<String, PropertyMetadata<?>> supportedProperties = connectorProperties.get(propertiesKey);
-        if (supportedProperties == null) {
-            throw new TrinoException(NOT_FOUND, formatPropertiesKeyForMessage(catalogNameForDiagnostics, propertiesKey) + " not found");
+        return doGetProperties(
+                propertiesKey,
+                catalogNameForDiagnostics,
+                separateProperties(propertiesKey, catalogNameForDiagnostics, properties).nonDefaultProperties,
+                session,
+                metadata,
+                accessControl,
+                parameters);
+    }
+
+    /**
+     * Separates the {@code Property}'s in {@code properties} based on whether they are set to DEFAULT or not
+     */
+    private SeparatedProperties separateProperties(
+            K propertiesKey,
+            String catalogNameForDiagnostics,
+            Iterable<Property> properties)
+    {
+        Map<String, Expression> nonDefaultProperties = new HashMap<>();
+        Set<String> defaultPropertyNames = new HashSet<>();
+        for (Property property : properties) {
+            String name = property.getName().getValue().toLowerCase(ENGLISH);   // property names are case-insensitive and normalized to lower case
+            if (nonDefaultProperties.containsKey(name) || defaultPropertyNames.contains(name)) {
+                throw new TrinoException(
+                        DUPLICATE_PROPERTY,
+                        format("%s %s property '%s' is specified more than once",
+                                formatPropertiesKeyForMessage(catalogNameForDiagnostics, propertiesKey),
+                                propertyType,
+                                name));
+            }
+            if (!property.isSetToDefault()) {
+                nonDefaultProperties.put(name, property.getNonDefaultValue());
+            }
+            else {
+                defaultPropertyNames.add(name);
+            }
+        }
+        return new SeparatedProperties(nonDefaultProperties, defaultPropertyNames);
+    }
+
+    /**
+     * Represents the result of separating a collection of {@code Property}'s based on whether they are set to DEFAULT or not. This is the
+     * return type of {@code separateProperties}.
+     */
+    private static class SeparatedProperties
+    {
+        final Map<String, Expression> nonDefaultProperties;
+        final Set<String> defaultPropertyNames;
+
+        SeparatedProperties(Map<String, Expression> nonDefaultProperties, Set<String> defaultPropertyNames)
+        {
+            this.nonDefaultProperties = ImmutableMap.copyOf(nonDefaultProperties);
```

Comment:
`ImmutableMap.copyOf(requireNotNull(nonDefaultProperties, "nonDefaultProperties is null"));` (similar below)

Reply:
I don't think that is necessary since `SeparatedProperties` is private. But anyway, I have added the null-checking code as you suggested.